### PR TITLE
feat: product detail page routing

### PR DESCRIPTION
**Pull request by pankaj-prog**
feat: product detail page routing
fix: scroll to top on location change

### CODE REVIEW

**Review by aiman-ansari — APPROVED** [review]
> Variables names are easy to understand .

> Looks good to merge :)

File: src/pages/CourseDetails/CourseDetails.jsx [hunk 1]
```diff
@@ -24,49 +26,86 @@ const course = {
...
-              <span>Content</span>
-              <span className="h5">{course.duration}</span>
+  const { productId } = useParams();
+  const [product, setProduct] = useState(null);
+  const [message, setMessage] = useState("loading....");
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(`/api/products/${productId}`);
```
Comment (m26unkwn) [inline src/pages/CourseDetails/CourseDetails.jsx:36]: Everything is good and readable.

If you can destruct the response here use it .`

const {data:{product}} = fetch Call`

**Review by m26unkwn — APPROVED** [review]

File: src/components/ScrollToTop/ScrollToTop.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React, { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+
+const ScrollToTop = ({ children }) => {
+  const location = useLocation();
+  useEffect(() => {
```
Comment (shruthipatil11) [inline src/components/ScrollToTop/ScrollToTop.jsx:6]: code looks good to me

**Review by shruthipatil11 — APPROVED** [review]

**Review by SinchanaMS — APPROVED** [review]
Looks good, Pankaj. The UI looks so clean!

A suggestion if you want to consider - "Clear" looks slightly out of place on your product listing page, probably because the width of the rest of the elements is shorter. It might look better if it's slightly positioned to the left or if you create a visual difference between the page and the filter pane.
 
image

**Pull request merged**